src/Sylius/Behat/Context/Setup/CustomerContext.php

```diff
@@ -0,0 +1,157 @@
+    ) {
+        $this->sharedStorage = $sharedStorage;
+        $this->customerRepository = $customerRepository;
+        $this->customerFactory = $customerFactory;
+        $this->userFactory = $userFactory;
+    }
+
+    /**
+     * @Given the store has customer :name with email :email
+     */
+    public function theStoreHasCustomerWithNameAndEmail($name, $email)
+    {
+        $partsOfName = explode(' ', $name);
+        $this->createCustomer($email, $partsOfName[0], $partsOfName[1]);
+    }
+
+    /**
+     * @Given the store has customer :email
+     */
+    public function theStoreHasCustomer($email)
+    {
+        $this->createCustomer($email, self::DEFAULT_CUSTOMER_FIRST_NAME, self::DEFAULT_CUSTOMER_LAST_NAME);
+    }
+
+    /**
+     * @Given the store has customer :email with first name :firstName
+     */
+    public function theStoreHasCustomerWithFirstName($email, $firstName)
+    {
+        $this->createCustomer($email, $firstName, self::DEFAULT_CUSTOMER_LAST_NAME);
+    }
+
+    /**
+     * @Given the store has customer :email with last name :lastName
+     */
+    public function theStoreHasCustomerWithLastName($email, $lastName)
+    {
+        $this->createCustomer($email, self::DEFAULT_CUSTOMER_FIRST_NAME, $lastName);
+    }
+
+    /**
+     * @Given there is disabled customer account :email with password :password
+     */
+    public function thereIsDisabledCustomerAccountWithPassword($email, $password)
+    {
+        $this->createCustomerAccount($email, $password, false);
+    }
+
+    /**
+     * @Given there is enabled customer account :email with password :password
+     */
+    public function theStoreHasEnabledCustomerAccountWithPassword($email, $password)
+    {
+        $this->createCustomerAccount($email, $password, true);
+    }
+
+    /**
+     * @param string $email
+     * @param string $firstName
+     * @param string $lastName
+     */
+    private function createCustomer($email, $firstName, $lastName)
+    {
+        $customer = $this->customerFactory->createNew();
+
+        $customer->setFirstName($firstName);
+        $customer->setLastName($lastName);
+        $customer->setEmail($email);
+
+        $this->sharedStorage->set('customer', $customer);
+        $this->customerRepository->add($customer);
+    }
+
+    /**
+     * @param string $email
+     * @param string $password
+     * @param bool $enabled
+     */
+    private function createCustomerAccount($email, $password, $enabled = true)
+    {
+        $user = $this->userFactory->createNew();
+        $customer = $this->customerFactory->createNew();
+
+        $customer->setFirstname(self::DEFAULT_CUSTOMER_FIRST_NAME);
+        $customer->setLastname(self::DEFAULT_CUSTOMER_LAST_NAME);
+        $customer->setEmail($email);
+
+        $user->setUsername($email);
+        $user->setPlainPassword($password);
+        $user->setEnabled($enabled);
+        $user->setCustomer($customer);
+
+        $customer->setUser($user);
```

Comment:
`$user->setCustomer($customer)` method already sets user on a customer so this line is redundant.